### ISSUE TITLE
Cut and paste list item into a new block creates an invalid paragraph block

### ISSUE THREAD

**Issue by bobbingwide** [WordPress/gutenberg#32467]
## Description
I was trying to cut and paste list items from one list to another. After saving and reediting the new list items were broken.

## Step-by-step reproduction instructions
1. Create a post with a list block with multiple list items.
2. Select a list item and copy it to the clipboard
3. Delete the selected item
4. Paste the clipboard content into the post
5. Save
6. View
7. Edit
8. The newly created block in step 4 is now broken
9. Attempt block recovery loses the list item's text.

## Expected behaviour
When the list item is pasted then it should become a list block.

## Actual behaviour
When the list item is pasted it becomes a list item in a paragraph block.
```
<!-- wp:paragraph -->
<p> <li>What is silver side reindeering?</li>  </p>
<!-- /wp:paragraph -->
```

## Screenshots or screen recording (optional)
result after step 7.
![image](https://user-images.githubusercontent.com/2474435/120858151-450b9100-c57a-11eb-8f91-0ad2909f93ca.png)

result after step 4. having gone back to step 2. to try again.
This shows the list item being created in a paragraph block.
![image](https://user-images.githubusercontent.com/2474435/120858795-32de2280-c57b-11eb-8d31-3267a4f2f3b9.png)

## WordPress information
- WordPress version:  5.7.2
- Gutenberg version: 10.7.1 
- Are all plugins except Gutenberg deactivated? No 
- Are you using a default theme (e.g. Twenty Twenty-One)?  No 

## Device information
- Device: Desktopn
- Operating system:  Windows 10 
- Browser:  Chrome

**Comment by bobbingwide** [comment]
For a demonstration of the problem plus a few other feechurs of the list block and block transforms see this video
https://youtu.be/FMe-pNfP4HQ

**Comment by skorasaurus** [comment]
I'm able to reproduce this; compiled from Gutenberg trunk earlier today (2021-06-30); 
notably, if you cut the list item instead of copy; text will transform into a paragraph (without any list formatting)  upon paste.

(sorry if you mention this in the video, i did not watch it).

**Comment by skorasaurus** [comment]
very similar examples of this cropping up recently: 
https://github.com/WordPress/gutenberg/issues/32467
https://github.com/WordPress/gutenberg/issues/32980
https://github.com/WordPress/gutenberg/issues/31067

**Comment by pbiron** [comment]
Similar (and probably related), in 5.8-RC4:

1. copy a list item to the clipboard
2. insert a heading block somewhere else
3. paste the clipboard into the heading block

That results in an empty heading block and a new list block, although neither block is "broken").

**Comment by annezazu** [comment]
I tried replicating this with 6.3 RC1 but was unable to. Closing this out but opening a new issue around the copy/paste experience with the new list block.
